core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -562,6 +543,8 @@ object ConfigCommand extends Logging {
           adminClient.describeCluster(new DescribeClusterOptions()).nodes().get().asScala.map(_.idString).toSeq :+ BrokerDefaultEntityName
         case ConfigType.CLIENT_METRICS =>
           adminClient.listClientMetricsResources().all().get().asScala.map(_.name).toSeq
+        case ConfigType.GROUP =>
+          adminClient.listConsumerGroups(new ListConsumerGroupsOptions().withTypes(Seq(GroupType.CONSUMER).toSet.asJava)).all.get.asScala.map(_.groupId).toSeq
```

Comment:
This is interesting. Because you've used `withTypes(GroupType.CONSUMER)`, this only lists out the consumers which are using the new consumer group protocol. So, I can set configs for other consumers, but `--describe` without a group name will not show them.

Really, the solution here I think is in KIP-1043 which adds `adminClient.listGroups()` so that all groups regardless of type can be listed. Once that KIP is approved, I'll improve this area. For now, I think `Seq(GroupType.CONSUMER, GroupType.CLASSIC)` is probably the best you can do. That does at least list out the configs for all new and classic consumer groups.

Reply:
> So, I can set configs for classic consumer groups, but --describe without a group name will not show them.

does `Share`  have similar issue? 

 > Once that KIP is approved, I'll improve this area. 

our of curiosity. I feel your suggestion `Seq(GroupType.CONSUMER, GroupType.CLASSIC)` is good enough if we only care about the configs of `CONSUMER` and `CLASSIC` . what is the improvement of this case introduced by KIP-1043?

Reply:
@chia7712 Group configs can be applied to all groups. KIP-932 introduces 5 new group configs which I expect to deliver in the coming days. For example, isolation level for a share group is a group-level config configured with `group.share.isolation.level`.

KIP-1043 would enable you to list groups of all types in a cleaner way because it introduces `AdminClient.listGroups()`. The thing I struggle with here is that `adminClient.listConsumerGroups()` sounds like it actually returns groups of all kinds, but that's misleading. The result might contain groups which are not consumer groups and if you attempt consumer group operations on them, you'll get errors. It should return consumer groups.

Obviously, there's some history here and the transition as the new consumer group protocol and the new group types are introduced has a few rough edges because these were not ideas that were anticipated a few years ago.